src/dmd/expression.d

```diff
@@ -4818,7 +4827,7 @@ extern (C++) final class DotVarExp : UnaExp
                                 /* checkModify will consider that this is an initialization
                                  * of v while it is actually an assignment of a field of v
                                  */
-                                scope modifyLevel = v.checkModify(loc, sc, dve.e1, !onlyUnion ? (flag | 2) : flag);
+                                scope modifyLevel = v.checkModify(loc, sc, dve.e1, !onlyUnion ? (flag | ModifyFlags.fieldAssign) : flag);
```

Comment:
Should the callers of `checkModifiable` be updated as well to pass `ModifyFlags.noError`?  In which case, the `flag` variable could also be switched over to the enum type.

Reply:
Done. I'm surprised how little the flag is used. I had to modify a single call site; the rest just go with the default.